File: wrappers/mvc/src/Kendo.Mvc/UI/Chart/ChartPane.cs
namespace Kendo.Mvc.UI
{
    public class ChartPane
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ChartPane" /> class
        /// with the specified name.
        /// </summary>
        /// <param name="name">The unique pane name.</param>
        public ChartPane(string name)
            : this()
        {
            Name = name;
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="ChartPane" /> class.
        /// </summary>
        public ChartPane()
        {
            Title = new ChartTitle();
            Margin = new ChartSpacing();
            Padding = new ChartSpacing();
            Border = new ChartElementBorder();
        }

        /// <summary>
        /// Gets or sets the unique pane name
        /// </summary>
        public string Name
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the pane height in pixels.
        /// </summary>
        public int? Height
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the pane title.
        /// </summary>
        public ChartTitle Title
        {
            get;
            private set;
        }

        /// <summary>
        /// Gets or sets the pane margin
        /// </summary>
        public ChartSpacing Margin
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the pane background color
        /// </summary>
        public string Background
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the pane padding
        /// </summary>
        public ChartSpacing Padding
        {
            get;
            set;
        }

        /// <summary>
        /// Gets or sets the pane border
        /// </summary>
        public ChartElementBorder Border
        {
            get;
            set;
        }

        public IChartSerializer CreateSerializer()
        {
            return new ChartPaneSerializer(this);
        }
    }
}

File: wrappers/php/lib/Kendo/Dataviz/UI/ChartPane.php
<?php

namespace Kendo\Dataviz\UI;

class ChartPane extends \Kendo\SerializableObject {
//>> Properties

    /**
    * The background color of the chart pane. Accepts a valid CSS color string, including hex and rgb.
    * @param string $value
    * @return \Kendo\Dataviz\UI\ChartPane
    */
    public function background($value) {
        return $this->setProperty('background', $value);
    }

    /**
    * The border of the chart pane.
    * @param \Kendo\Dataviz\UI\ChartPaneBorder|array $value
    * @return \Kendo\Dataviz\UI\ChartPane
    */
    public function border($value) {
        return $this->setProperty('border', $value);
    }

    /**
    * Specifies whether the charts in the pane should be clipped. By default all charts except radar, polar and 100% stacked charts are clipped.
    * @param boolean $value
    * @return \Kendo\Dataviz\UI\ChartPane
    */
    public function clip($value) {
        return $this->setProperty('clip', $value);
    }

    /**
    * The chart pane height in pixels.
    * @param float $value
    * @return \Kendo\Dataviz\UI\ChartPane
    */
    public function height($value) {
        return $this->setProperty('height', $value);
    }

    /**
    * The margin of the pane. A numeric value will set all margins.
    * @param float|\Kendo\Dataviz\UI\ChartPaneMargin|array $value
    * @return \Kendo\Dataviz\UI\ChartPane
    */
    public function margin($value) {
        return $this->setProperty('margin', $value);
    }

    /**
    * The unique name of the chart pane.
    * @param string $value
    * @return \Kendo\Dataviz\UI\ChartPane
    */
    public function name($value) {
        return $this->setProperty('name', $value);
    }

    /**
    * The padding of the pane. A numeric value will set all paddings.
    * @param float|\Kendo\Dataviz\UI\ChartPanePadding|array $value
    * @return \Kendo\Dataviz\UI\ChartPane
    */
    public function padding($value) {
        return $this->setProperty('padding', $value);
    }

    /**
    * The title configuration of the chart pane.
    * @param string|\Kendo\Dataviz\UI\ChartPaneTitle|array $value
    * @return \Kendo\Dataviz\UI\ChartPane
    */
    public function title($value) {
        return $this->setProperty('title', $value);
    }

//<< Properties
}

?>
